lib/xandra/authenticator.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Xandra.Authenticator do
+  @moduledoc """
+  A behaviour module for implementing a Cassandra authenticator
+
+  ## Example
+
+    defmodule MyAuthenticator do
+      @behaviour Xandra.Authenticator
+
+      def response_body(options) do
+        [Keyword.fetch(options, :foo), Keyword.fetch(options, :bar)]
+      end
+    end
+
+    # To use your authenticator
+
+    Xandra.start_link(authentication: {MyAuthenticator, foo: "foo", bar: "bar"}
+
+  Xandra supports Cassandra's PasswordAuthenticator by default, see
+  `Xandra.Authenticator.Password` for more information.
+  """
+
+  @doc """
+  Returns an IOdata that's used as the response body to Cassandra server's auth challenge
```

Comment:
1. We don't capitalize iodata like that. You can just use "iodata". :)
1. "to Cassandra server's auth challenge..." -> "to Cassandra's auth challenge..." IMO.
1. Missing trailing period.